datafusion/expr/src/expr.rs

```diff
@@ -459,22 +482,40 @@ pub struct WindowFunction {
 }
 
 impl WindowFunction {
-    /// Create a new Window expression
-    pub fn new(
-        fun: window_function::WindowFunction,
-        args: Vec<Expr>,
-        partition_by: Vec<Expr>,
-        order_by: Vec<Expr>,
-        window_frame: window_frame::WindowFrame,
-    ) -> Self {
+    /// Create a new Window expression with the specified argument an
+    /// empty `OVER` clause
+    pub fn new(fun: impl Into<window_function::WindowFunction>, args: Vec<Expr>) -> Self {
```

Comment:
This is the major change -- so that WindowFunction acts like a `Builder` which can be used to set `partition by`, `order by` and `window_frame` with a more self documenting style